### PR TITLE
[3.5] CVE-2023-47108: Backport go.opentelemetry.io/otel@v1.20.0 and go.open…

### PR DESCRIPTION
**Pull request by sharathsivakumar**
This PR addresses the https://github.com/etcd-io/etcd/issues/16926 by bumping dependencies:
- bump dependency go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc from 0.45.0 to 0.46.0

### CODE REVIEW

File: bill-of-materials.json [hunk 1]
```diff
@@ -188,6 +206,15 @@
 			}
 		]
 	},
+	{
+		"project": "github.com/grpc-ecosystem/grpc-gateway/v2",
```
Comment (ahrtr) [inline bill-of-materials.json:210]: We don't bump grpc-gateway in 3.5. It's a big change. 

https://github.com/etcd-io/etcd/pull/16595

Comment (ahrtr) [reply bill-of-materials.json:210]: It should be fine to just introduce an indirect dependency on grpc-gateway/v2. It means we are still using grpc-gateway v1 in 3.5

**Comment by ahrtr** [comment]
Please refer to https://github.com/etcd-io/etcd/pull/16454/commits/40c7f72b1d62790554a0e43cd325295d034deba0 to resolve the workflow failure in `linux-amd64-fmt`.

**Comment by sharathsivakumar** [comment]
@ahrtr thanks for the inputs. I will work on the fixes and push the changes.

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks

**Comment by ahrtr** [comment]
cc @jmhbnz @serathius @wenjiaswe @fuweid @mitake

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @sharathsivakumar

**Review by fuweid — APPROVED** [review]
LGTM

**Comment by fuweid** [comment]
We still need changelog's change in main branch after merging this one :)

**Comment by ahrtr** [comment]
> We still need changelog's change in main branch after merging this one :)

Yes, @sharathsivakumar Please feel free to raise a separate PR to change the changelog. thx

**Pull request merged**